File: Golang/juliaSet.go
package main

import (
    "image"
    "image/color"
    "image/png"
    "math/cmplx"
    "os"
)

const (
    width       = 800
    height      = 800
    maxIter     = 300
    xMin, xMax  = -1.5, 1.5
    yMin, yMax  = -1.5, 1.5
    cRe, cIm    = -0.7, 0.27015
)

func main() {
    img := image.NewRGBA(image.Rect(0, 0, width, height))

    for x := 0; x < width; x++ {
        for y := 0; y < height; y++ {
            zx := xMin + (xMax-xMin)*float64(x)/width
            zy := yMin + (yMax-yMin)*float64(y)/height
            z := complex(zx, zy)
            c := complex(cRe, cIm)
            var i int
            for i = 0; i < maxIter; i++ {
                if cmplx.Abs(z) > 2 {
                    break
                }
                z = z*z + c
            }
            colorVal := uint8(255 * i / maxIter)
            col := color.RGBA{colorVal, colorVal, colorVal, 255}
            img.Set(x, y, col)
        }
    }

    file, err := os.Create("juliaSet.png")
    if err != nil {
        panic(err)
    }
    defer file.Close()

    if err := png.Encode(file, img); err != nil {
        panic(err)
    }
}

File: Python/juliaSet.py
import numpy as np
"""
This script generates and plots a Julia set using the given parameters.
Parameters:
    width (int): The width of the output image in pixels.
    height (int): The height of the output image in pixels.
    x_min (float): The minimum x-coordinate value for the plot.
    x_max (float): The maximum x-coordinate value for the plot.
    y_min (float): The minimum y-coordinate value for the plot.
    y_max (float): The maximum y-coordinate value for the plot.
    max_iterations (int): The maximum number of iterations to perform.
    c (complex): The complex constant used in the Julia set formula.
Variables:
    x (ndarray): Array of x-coordinates.
    y (ndarray): Array of y-coordinates.
    X (ndarray): 2D grid of x-coordinates.
    Y (ndarray): 2D grid of y-coordinates.
    Z (ndarray): 2D grid of complex numbers.
    iterations (ndarray): Array to store the iteration count for each point.
The script performs the iteration to generate the Julia set and plots the result using matplotlib.
Note:
    Other color maps that can be used include 'viridis', 'plasma', 'magma', 'cividis', etc.
"""
import matplotlib.pyplot as plt

# Parameters for the Julia set
width, height = 800, 800
x_min, x_max = -1.5, 1.5
y_min, y_max = -1.5, 1.5
max_iterations = 300
c = complex(-0.7, 0.27015)

# Create a grid of complex numbers
x = np.linspace(x_min, x_max, width)
y = np.linspace(y_min, y_max, height)
X, Y = np.meshgrid(x, y)
Z = X + 1j * Y

# Initialize the iteration count array
iterations = np.zeros(Z.shape, dtype=int)

# Perform the iteration
for i in range(max_iterations):
    mask = np.abs(Z) < 4
    Z[mask] = Z[mask] ** 2 + c
    iterations[mask] = i

# Plot the Julia set
plt.imshow(iterations, extent=(x_min, x_max, y_min, y_max), cmap='plasma', origin='lower')
plt.colorbar()
plt.title('Julia Set')
plt.xlabel('Re')
plt.ylabel('Im')
plt.show()